packages/mana/persistableevent.go

```diff
@@ -0,0 +1,144 @@
+package mana
+
+import (
+	"crypto/sha256"
+	"math"
+	"time"
+
+	"github.com/iotaledger/goshimmer/dapps/valuetransfers/packages/transaction"
+	"github.com/iotaledger/hive.go/identity"
+	"github.com/iotaledger/hive.go/marshalutil"
+	"github.com/iotaledger/hive.go/objectstorage"
+)
+
+// PersistableEvent is a persistable pledge event.
+type PersistableEvent struct {
+	objectstorage.StorableObjectFlags
+	Type          byte // pledge or revoke
+	NodeID        identity.ID
+	Amount        float64
+	Time          time.Time
+	ManaType      Type // access or consensus
+	TransactionID transaction.ID
+	bytes         []byte
+}
+
+// Bytes marshals the persistable event into a sequence of bytes.
+func (p *PersistableEvent) Bytes() []byte {
+	if bytes := p.bytes; bytes != nil {
+		return bytes
+	}
+	// create marshal helper
+	marshalUtil := marshalutil.New()
+	marshalUtil.WriteByte(p.Type)
+	marshalUtil.WriteInt64(int64(p.ManaType))
+	marshalUtil.WriteBytes(p.NodeID.Bytes())
+	marshalUtil.WriteTime(p.Time)
+	marshalUtil.WriteBytes(p.TransactionID.Bytes())
+	marshalUtil.WriteUint64(math.Float64bits(p.Amount))
+	p.bytes = marshalUtil.Bytes()
+	return p.bytes
+}
+
+// Update updates the event in storage.
+func (p *PersistableEvent) Update(objectstorage.StorableObject) {
+	panic("should not be updated")
+}
+
+// ObjectStorageKey returns the key of the persistable mana.
+func (p *PersistableEvent) ObjectStorageKey() []byte {
+	return []byte(p.TransactionID.String() + p.Time.String() + p.NodeID.String())
```

Comment:
```p.TransactionID.String() + p.Time.String() + p.NodeID.String()``` is not a unique key, as a `Pledge` and a `Revoke` event that were caused by the same tx to happen, at the same time, for the same node would resolve to the same storage key. This might happen when I just move my funds to a different address, but from the same node.

Reply:
you are right.